E1721. Use Ajax for adding participants, TA, Edit Questionnaires screens, Unsubmit Reviews

psvaidya opened the pull request:
With these commits, an instructor will be able to add a new Participant to an assignment and also add/remove TA from a course remotely using ajax without reloading the page. 
Also, on "Assign Reviewers" page for assignments, the status of the review (Assigned, Saved, Submitted) will be seen for each review. If status is submitted, there will be a link to unsubmit the review. Clicking on this will unsubmit the review, and status will be changed to Saved, remotely using ajax without reloading the page.

Details of changes done are on this wiki page: http://wiki.expertiza.ncsu.edu/index.php/CSC/ECE_517_Spring_2017/Use_Ajax_to_add_Participants,TA_and_Edit_Questionnaire

Winbobob commented:
Hi team,
It seems that only one test is failed, please fix it if you can.
For issues reported by code climate, please fix them except `Assignment Branch Condition size`.

Thanks,
Zhewei

psvaidya commented:
We have corrected the failing test and included more tests.
We have also fixed all the codeclimate except "Assignment Branch Condition size" as mentioned by Zhewei. Please approve this remaining issue related to "Assignment Branch Condition size"

psvaidya commented:
We have corrected the failing test and included more tests.
We have also fixed all the codeclimate except "Assignment Branch Condition size" as mentioned by Zhewei. Please approve this remaining issue related to "Assignment Branch Condition size"
